### PR TITLE
template: isoweek support

### LINKED ISSUE #2877: Request to support ISO8601 week numbers
In syslog-ng (3.20.1), I find it convenient to incorporate the week number as part of a file template:

```
file("/var/log/${FACILITY}-${YEAR}W${WEEK}.log");
```

The problem is that the numbering does not conform to ISO 8601. Syslog-ng considers week 01 to be that in which the first Monday falls. However, the ISO standard week numbering system entails that it is the first Thursday instead.

Hence, at the time of writing this post, syslog-ng considers the current day (2019-08-12) to be in week 32 but it is actually 33. This can be demonstrated with the cal utility that ships with util-linux. Note that the -m option is used below to coerce the ISO scheme, which requires that Monday is the first day of the week. Otherwise, it is dependent on the current locale and may use a scheme that is not ISO-conformant e.g. if LC_TIME="en_US.utf8" in Linux.

```
# cal -w -m
      August 2019      
   Mo Tu We Th Fr Sa Su
31           1  2  3  4
32  5  6  7  8  9 10 11
33 12 13 14 15 16 17 18
34 19 20 21 22 23 24 25
35 26 27 28 29 30 31

# cal -w -m 1 2019
      January 2019     
   Mo Tu We Th Fr Sa Su
 1     1  2  3  4  5  6
 2  7  8  9 10 11 12 13
 3 14 15 16 17 18 19 20
 4 21 22 23 24 25 26 27
 5 28 29 30 31
```

My request is to add an ISOWEEK macro that conforms to the ISO numbering system.

(EDIT: fixed typo in env var)

### PR DESCRIPTION
**Pull request by Kokan**
There was a request to support ISO week numbering (the current `WEEK` macro is a naive week number calculation).

The new macros: `${ISOWEEK}` and work with timestamp prefixes: `P_, S_, R_, C_`.

Fixes #2877

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by alltilla** [comment]
Can you please share the source of the calculations? I found a Wikipedia page about "ISO 8601 week date", but I would like to check your source, too.

**Comment by Kokan** [comment]
Sure, thanks for asking. Added with mit license to the source file.

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: lib/timeutils/wallclocktime.c [hunk 1]
```diff
@@ -52,6 +52,31 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
+/* MIT: musl http://git.musl-libc.org/cgit/musl/tree/src/time/strftime.c?id=6ad514e4e278f0c3b18eb2db1d45638c9af1c07f#n19 */
+/*
+ *
+ *  Copyright © 2005-2019 Rich Felker, et al.
+ *
```
[the remark this answers is not in the document]
Comment (Kokan) [reply lib/timeutils/wallclocktime.c]: done, please check.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: lib/template/tests/test_macro.c [hunk 2]
```diff
@@ -102,6 +102,24 @@ Test(macro, test_date_iso_week_last_iso_week)
   g_string_free(resolved, TRUE);
 }
 
+Test(macro, test_date_iso_week_previouse_week_if_thuesday_is_not_in_it)
```
Comment (furiel) [inline lib/template/tests/test_macro.c]: Just tuesday.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**